### PR TITLE
Fix kalman.c

### PR DESCRIPTION
**Pull request by Quick-Flash**
Some issues with this kalman filter were brought to my attention. I have since been playing with it and getting it fixed. So far this works for me. I'll be having more people test it and see what they think as well, but it should make this filter more similar to the filter in the helio fc which should be for the better me thinks. Pushing this again as the last one had stick pids in it as well, which I did not want.

### CODE REVIEW

File: src/main/common/kalman.c [hunk 1]
```diff
@@ -116,17 +116,15 @@ FAST_CODE float kalman_process(kalman_t* kalmanState, float input, float target)
   //update last state
   kalmanState->lastX = kalmanState->x;
 
+  if (kalmanState->lastX != 0.0f) {
```
Comment (gretel) [inline src/main/common/kalman.c]: guess it's fine but try to have the most common case first - for performance reasons :)

Comment (Quick-Flash) [reply src/main/common/kalman.c]: Yea that isn't a very common case at all, but how would you suggest I change it, and still maintain its functionality?

Comment (gretel) [reply src/main/common/kalman.c]: try to use `==` whenever possible and match the most probable case first. guess it's fine for now

Comment (Quick-Flash) [reply src/main/common/kalman.c]: I mean in this code this is added just to prevent a divide by 0 error. This should be good to merge.

**Comment by nerdCopter** [comment]
the internets show "compliancy" to be:
```C
  if (kalmanState->lastX == 0.0f)  {
    /* Handle error */
  } else {
    /* Do the work */
  }
```
but also has more complex solutions like error trapping: http://www.keil.com/support/man/docs/ARMCC/armcc_chr1359124235078.htm
Apparently C++ has better div by zero handling over C (NaN/Infinity)

File: src/main/common/kalman.c (hunk 1 above)
Comment (gretel) [reply src/main/common/kalman.c]: i wasnt saying it is not. at the time of my previous writing the status checks were still running. you should be open to suggestions though and not just pushing for approval.

**Comment by gretel** [comment]
dang i clicked the wrong button. hang on travis will just taker another day or so to build :(

**Pull request merged**